roxie/roxiemem/roxiemem.cpp

```diff
@@ -4459,18 +4463,34 @@ class CChunkingRowManager : public CRowManager
         }
 
         DataBuffer *dfinger = activeBuffs;
+        unsigned gone = 0;
+        unsigned dead = 0;
         while (dfinger)
         {
             DataBuffer *next = dfinger->nextActive;
-            if (memTraceLevel >= 2 && dfinger->queryCount()!=1)
+            if (memTraceInconsistencies && getSeqFromDataId((memsize_t)next) != 0)
+                logctx.CTXLOG("INTERNAL ERROR: unexpected sequence number in nextActive in ~CChunkingRowManager - addr=%p rowMgr=%p",
+                        dfinger, this);
+
+            unsigned count = dfinger->queryCount();
+            if (count == 1)
+                gone++;
+            else if (count == 0)
+                dead++;
+
+            if (memTraceLevel >= 2 && count!=1)
                 logctx.CTXLOG("RoxieMemMgr: Memory leak: %d records remain linked in active dataBuffer list - addr=%p rowMgr=%p", 
                         dfinger->queryCount()-1, dfinger, this);
-            dfinger->nextActive = 0;
-            dfinger->mgr = NULL; // Avoid calling back to noteDataBufferReleased, which would be unhelpful
+            dfinger->nextActive = nullptr;
+            dfinger->mgr = nullptr; // Avoid calling back to noteDataBufferReleased, which would be unhelpful
             dfinger->count.store(0, std::memory_order_relaxed);
             dfinger->released();
             dfinger = next;
         }
+
+        if ((dataBuffs != possibleGoers.load()) || (gone != possibleGoers))
```

Comment:
Perhaps also add && if (memTraceInconsistencies) to the logging here ?

Reply:
I think this should only happen once per query, and be very unlikely - so I think it is ok to trace it always.  The other tracing could be generated many times per query.

Reply:
ok.
Approved.